sync/models/ir_model.py

```diff
@@ -0,0 +1,39 @@
+# _update_selection
+
+from odoo import fields, models
+
+
+class IrModelSelection(models.Model):
+    _inherit = 'ir.model.fields.selection'
+
+    def _update_sync_project_context(self):
+        spc = self.env['sync.project.context']
+        many2many_values = spc.search([]).mapped(lambda v: v.name)
+        new_selection_values = self.env['ir.model.fields.selection'].search(
+            [('field_id.model', '=', 'sync.project'), ('field_id.name', '=', 'eval_context')]
+        ).filtered(lambda v, vals=many2many_values: v.value not in vals)
+        if new_selection_values:
+            for new_selection_value in new_selection_values:
+                spc.create({
+                    'name': new_selection_value.value,
+                    'display_name': new_selection_value.name
+                })
+
+    def _reflect_selections(self, model_names):
+        """
+            Temporary compatibility helper for modules still using eval_context's selection_add.
+            This field is being deprecated. Please create sync.project.context records instead.
+            See data/sync_project_context.xml
+        """
+        res = super(IrModelSelection, self)._reflect_selections(model_names)
```

Comment:
I don't like idea of making such deep changes. Anyway, this code is not needed, since I don't plan to support old implementations